### PR TITLE
support autoround hpu format

### PR DESCRIPTION
**Pull request by yintong-lu**
changes on run_lyt.py and evaluation.py are temporary, and would be reverted after tests.

### CODE REVIEW

File: auto_round/auto_quantizer.py [hunk 1]
```diff
@@ -148,7 +148,11 @@ def from_config(cls, quantization_config: Union[QuantizationConfigMixin, Dict],
         if "auto-round" in quant_method:
```
Comment (wenhuach21) [inline auto_round/auto_quantizer.py]: move the below to here,  if "auto-round" in quant_method or supported_hpu

File: auto_round/auto_quantizer.py [hunk 2]
```diff
@@ -148,7 +148,11 @@ def from_config(cls, quantization_config: Union[QuantizationConfigMixin, Dict],
...
             target_cls = AutoRoundQuantizer
         else:
-            target_cls = AUTO_QUANTIZER_MAPPING[quant_method]
+            from transformers.utils.import_utils import is_optimum_available
```
Comment (wenhuach21) [inline auto_round/auto_quantizer.py]: do not change anything here to support the original behaviour

File: auto_round/auto_quantizer.py [hunk 3]
```diff
@@ -303,7 +307,7 @@ def convert_model(self, model: nn.Module):
         quantization_config = model.config.quantization_config
         bits = quantization_config.bits
         group_size = quantization_config.group_size
-        data_type = quantization_config.data_type
+        data_type = quantization_config.data_type if hasattr(quantization_config, "data_type") else "int"
```
Comment (wenhuach21) [inline auto_round/auto_quantizer.py]: why changing here

Comment (yintong-lu) [reply auto_round/auto_quantizer.py]: some gptq models like qwen2-1.5b-int4-inc does not have this attribute.

File: README.md [hunk 4]
```diff
@@ -168,6 +168,23 @@ inputs = tokenizer(text, return_tensors="pt").to(model.device)
 print(tokenizer.decode(model.generate(**inputs, max_new_tokens=50)[0]))
 ```
 
+### Intel Gaudi-2
+
+```python
+##pip install auto-gptq
+from transformers import AutoModelForCausalLM, AutoTokenizer
+##from auto_round.auto_quantizer import AutoHfQuantizer ## uncomment it for models with auto_round format
```
Comment (wenhuach21) [inline README.md]: uncomment it

File: auto_round/auto_quantizer.py [hunk 5]
```diff
@@ -106,6 +106,12 @@ def is_autoround_exllamav2_available():
         res = False
     return res
 
+def is_hpu_supported():
+    try:
+        import habana_frameworks.torch.core as htcore # pylint: disable=E0401
```
Comment (wenhuach21) [inline auto_round/auto_quantizer.py:112]: better check the version >=1.17?

File: auto_round_extension/hpu/qlinear_hpu.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        out = out + self.bias if self.bias is not None else out
+        return out
+
+    def unpack_zeros_from_cuda_old_format(self):
+        zeros = torch.bitwise_right_shift(
+            torch.unsqueeze(self.qzeros, 2).expand(-1, -1, 32 // self.bits),
+            self.wf.unsqueeze(0),
+        ).to(torch.int16 if self.bits == 8 else torch.int8)
+
+        # zeros = zeros + 1 # NOTE: Commenting this line might cause accuracy regression.
```
Comment (wenhuach21) [inline auto_round_extension/hpu/qlinear_hpu.py]: support different logic for different formats

File: examples/language-modeling/eval_042/evaluation.py [hunk 7]
```diff
@@ -595,6 +599,9 @@ def evaluate(
         quantization_config = config.quantization_config
         if "quant_method" in quantization_config and "auto-round" in quantization_config["quant_method"]:
             from auto_round.auto_quantizer import AutoHfQuantizer
+        elif "quant_method" in quantization_config and quantization_config["quant_method"] == "gptq":
+            if args.device == "hpu":
```
Comment (wenhuach21) [inline examples/language-modeling/eval_042/evaluation.py:603]: just check hpu available is fine?

File: README.md [hunk 8]
```diff
@@ -172,12 +172,13 @@ print(tokenizer.decode(model.generate(**inputs, max_new_tokens=50)[0]))
...
 ##pip install auto-gptq
+import torch
 from transformers import AutoModelForCausalLM, AutoTokenizer
-##from auto_round.auto_quantizer import AutoHfQuantizer ## uncomment it for models with auto_round format
+from auto_round.auto_quantizer import AutoHfQuantizer ## uncomment it for models with auto_round format
 import habana_frameworks.torch.core as htcore
 import habana_frameworks.torch.hpu as hthpu
 quantized_model_path = "./tmp_autoround"
-model = AutoModelForCausalLM.from_pretrained(quantized_model_path, device_map="auto").to('hpu')
+model = AutoModelForCausalLM.from_pretrained(quantized_model_path, device_map="auto").to('hpu').to(torch.float32)
```
Comment (wenhuach21) [inline README.md:180]: is there any issue with bfloat16?

File: README.md [hunk 9]
```diff
@@ -168,6 +168,24 @@ inputs = tokenizer(text, return_tensors="pt").to(model.device)
 print(tokenizer.decode(model.generate(**inputs, max_new_tokens=50)[0]))
 ```
 
+### Intel Gaudi-2
+
+```python
+##pip install auto-gptq
```
Comment (wenhuach21) [inline README.md]: is auto-gptq still a requirement?

File: README.md [hunk 10]
```diff
@@ -168,6 +168,24 @@ inputs = tokenizer(text, return_tensors="pt").to(model.device)
...
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from auto_round.auto_quantizer import AutoHfQuantizer ## uncomment it for models with auto_round format
```
Comment (wenhuach21) [inline README.md]: remove the comment

**Review by wenhuach21 — APPROVED** [review]

File: auto_round/auto_quantizer.py (hunk 5 above)
Comment (wenhuach21) [inline auto_round/auto_quantizer.py:112]: utils.py has  is_optimum_habana_available, please have a check

File: README.md (hunk 8 above)
Comment (yintong-lu) [reply README.md:180]: Yes, a neneric failure of 'dequantize_4_bit_f16_with synStatus_26' would occur.

**Pull request merged**